File: class03.py
def maxProduct(nums):
    # 請用你的程式補完這個函式的區塊
    #預設第一次for 外+內 跑出來的第一個數字
    temp = nums[0]*nums[1] 
    # data=nums[1]-nums[0]
    for i in range(len(nums)):
        
        # print("nums[i]:",nums[i])
        for j in range(i+1,len(nums)):
            # print("nums[j]:",nums[j])
            sum=nums[i]*nums[j]
            if sum>=temp:
                temp=sum
    print("MAX:",max(temp,sum))
        
# maxProduct([5, 20, 2, 6]) # 得到 120
maxProduct([10, -20, 0, 3]) # 得到 30
# maxProduct([-1, 2]) # 得到 -2
# maxProduct([-1, 0, 2]) # 得到 0

File: class03.js
function maxProduct(nums){
    // 請用你的程式補完這個函式的區塊
   let sum=Number.NEGATIVE_INFINITY;
   let max=Number.NEGATIVE_INFINITY;
   for(let i=0; i<nums.length;i++){
       console.log("外面的i:",i);
        for(let j=i+1;j<nums.length;j++){
            sum=nums[i]*nums[j]
            console.log('i:',i,'j:',j,'sum:',sum);
            max=Math.max(sum,max);
             }     
        }  
        console.log(max);
        return max;
};
  
// maxProduct([5, 20, 2, 6]) // 得到 120
maxProduct([10, -20, 0, 3]) // 得到 30
// maxProduct([-1, 2]) // 得到 -2
// maxProduct([-1, 0, 2]) // 得到 0

